File: client/src/main/java/com/skm/sftp/client/ClientScriptModule.java
package com.skm.sftp.client;
import com.jcraft.jsch.JSchException;
import com.jcraft.jsch.SftpException;
import com.skm.sftp.AbstractScriptModule;
import com.skm.sftp.SftpUtil;

public class ClientScriptModule extends AbstractScriptModule {
  @Override
  protected void uploadImpl(String remoteHost, String username, String password, String localPath, String remotePath) throws JSchException, SftpException {
   new SftpUtil(remoteHost, username, password, localPath, remotePath).uploadSshj();
  }

  @Override
  protected void downloadImpl(String remoteHost, String username, String password, String localPath, String remotePath) throws JSchException, SftpException {
    new SftpUtil(remoteHost, username, password, localPath, remotePath).downloadSshj();
  }
}

File: client/src/main/kotlin/com/skm/sftp/client/ClientScriptModule.kt
package com.skm.sftp.client

import com.inductiveautomation.ignition.client.gateway_interface.ModuleRPCFactory
import com.skm.sftp.AbstractScriptModule
import com.skm.sftp.SFTPBlackBox


class ClientScriptModule: AbstractScriptModule() {
    private val rpc: SFTPBlackBox = ModuleRPCFactory.create(
        "com.skm.sftp",
        SFTPBlackBox::class.java
    )

    override fun uploadImpl(
        remoteHost: String,
        username: String,
        password: String,
        localPath: String,
        remotePath: String
    ) {
        rpc.upload(remoteHost, username, password, localPath, remotePath)
    }


    override fun downloadImpl(
        remoteHost: String,
        username: String,
        password: String,
        localPath: String,
        remotePath: String
    ) {
        rpc.download(remoteHost, username, password, localPath, remotePath)
    }


}